### PR TITLE
Minor fix for RSAPSS/SHA384 support for signing of CBnT structures

### PR DESCRIPTION
**Pull request by ChriMarMe**
Signed-off-by: Christopher Meis <christopher.meis@9elements.com>

### CODE REVIEW

File: pkg/intel/metadata/manifest/signature_types.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ func NewSignatureData(
 	}
 	switch signAlgo {
 	case AlgRSAPSS:
+		fmt.Println("AlgRSAPSS invoked")
```
Comment (xaionaro) [inline pkg/intel/metadata/manifest/signature_types.go:49]: Garbage

File: pkg/intel/metadata/manifest/signature_types.go [hunk 2]
```diff
@@ -58,6 +59,7 @@ func NewSignatureData(
 			SaltLength: rsa.PSSSaltLengthAuto,
 			Hash:       crypto.SHA384,
 		}
+		fmt.Printf("Key: %v\nData: %v\nhashData: %v\n", rsaPrivateKey, signedData, bpmHash)
```
Comment (xaionaro) [inline pkg/intel/metadata/manifest/signature_types.go:62]: Garbage

File: pkg/intel/metadata/manifest/signature_types.go (hunk 1 above)
Comment (ChriMarMe) [reply pkg/intel/metadata/manifest/signature_types.go:49]: Yes I know, thats why the the commit is called: [Print stuff to understand what is going wrong](https://github.com/linuxboot/fiano/pull/363/commits/d74ebcebd92a2aac983ce0dae0728f78cf182def)

File: pkg/intel/metadata/manifest/signature_types.go (hunk 2 above)
Comment (ChriMarMe) [reply pkg/intel/metadata/manifest/signature_types.go:62]: Same as the other comment.

File: pkg/intel/metadata/manifest/signature_types.go [hunk 3]
```diff
@@ -194,19 +206,32 @@ func (s SignatureRSAASA) String() string {
...
+	switch hashAlgo {
+	case AlgSHA256:
+		hashfunc = crypto.SHA256
+	case AlgSHA384:
+		hashfunc = crypto.SHA384
+	default:
+		return fmt.Errorf("signature verification for RSAASA only supports SHA256 and SHA384")
+	}
+
+	err = rsa.VerifyPKCS1v15(pk, hashfunc, hash, s)
```
Comment (xaionaro) [inline pkg/intel/metadata/manifest/signature_types.go:234]: Couldn't we use the signature size to detect the hash algo if it is not defined (if it is AlgNULL)?

Comment (ChriMarMe) [reply pkg/intel/metadata/manifest/signature_types.go:234]: Spec says keysize (in bit)/ 8 = signature size. Looks like there is no way to derive hash algo from that information.

**Review by xaionaro — APPROVED** [review]

**Pull request merged**